### PR TITLE
allow new projects to be added

### PR DESCRIPTION
**Pull request by afeld**
Builds on #85.

/cc https://github.com/18F/concourse-compliance-testing/issues/75

### CODE REVIEW

File: pipelines/zap/lib/target_initializer.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @config = config
+    @targets = targets
+  end
+
+  def run
+    targets.each do |target|
+      name = target['name']
+      key = "results/#{name}.json"
+
+      # since Concourse resources stall on a `get` if the remote file doesn't exist, we need to seed the last-results
```
Comment (DavidEBest) [inline pipelines/zap/lib/target_initializer.rb:17]: This is really clever. 👍

**Comment by DavidEBest** [comment]
Is this ready for review?

File: pipelines/zap/lib/target_initializer.rb (hunk 1 above)
Comment (afeld) [inline pipelines/zap/lib/target_initializer.rb:17]: Clever, huge hack...same difference 😉

**Comment by afeld** [comment]
Yep, ready! Was waiting for #85 to be merged so the diff would be cleaner.

I fought for a bit with trying to handle non-existent `last-results` or Team API entries for new projects, both of which would have required custom resources. From [the Concourse Slack](https://concourseci.slack.com/archives/general/p1459809441005060):

> no one should be treating errors as a normal part of pipeline operation

Therefore, this pull request contains a couple of workarounds:
- Add a `skip_team_api` option for the `targets.json` entries, which will be useful eventually for supporting non-18F projects.
- Add a rake task to add empty JSON files for the results for all of the listed projects, which will allow new project jobs to run.

Closes #75 by avoiding the issue.

**Comment by DavidEBest** [comment]
LGTM. :shipit:

I have some concerns about adding configuration flags and mingling those flags with the actual target data, but I don't have a better solution. Maybe if we just write a custom resource... :trollface:

**Comment by afeld** [comment]
Heh. The problem is that it's tough to determine if the resource doesn't exist because it's not there yet, or if it's because there was a misspelling in the configuration or something. I say we merge this until we figure out something better.

Another option is to say "if the fields that we need are in the `targets.json` (right now just `links`), don't bother with fetching the data from the Team API." That may get harder to determine if we end up using more fields.

**Comment by DavidEBest** [comment]
I agree. Better to merge this and iterate.

**Pull request merged**